optimize/qa/schema-integrity-tests/src/test/java/io/camunda/optimize/test/upgrade/client/ElasticsearchSchemaTestClient.java

```diff
@@ -36,106 +36,50 @@
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.repositories.fs.FsRepository;
 
-public class ElasticClient {
+public class ElasticsearchSchemaTestClient extends AbstractDatabaseSchemaTestClient {
 
-  public static final String SNAPSHOT_REPOSITORY_NAME = "my_backup";
-  public static final String SNAPSHOT_NAME_1 = "snapshot_1";
-  public static final String SNAPSHOT_NAME_2 = "snapshot_2";
-  public static final String OPTIMIZE_INDEX_PREFIX = "optimize";
-  public static final String DEFAULT_OPTIMIZE_INDEX_PATTERN =
-      OPTIMIZE_INDEX_PREFIX + "-*,-" + OPTIMIZE_INDEX_PREFIX + "-update-log*";
   private static final org.slf4j.Logger log =
-      org.slf4j.LoggerFactory.getLogger(ElasticClient.class);
-  private static final String[] SETTINGS_FILTER =
-      new String[] {
-        "index.analysis.*",
-        "index.number_of_shards",
-        "index.number_of_replicas",
-        "index.max_ngram_diff",
-        "index.mapping.*",
-        "index.refresh_interval"
-      };
-  private final String name;
+      org.slf4j.LoggerFactory.getLogger(ElasticsearchSchemaTestClient.class);
   private final RestHighLevelClient client;
-  private final Boolean readonlySnapshotRepository;
-
-  public ElasticClient(
-      final String name,
-      final int port,
-      final String host,
-      final Boolean readonlySnapshotRepository) {
-    this.name = name;
+
+  public ElasticsearchSchemaTestClient(final String name, final int port) {
+    super(name);
     client =
         new RestHighLevelClient(
```

Comment:
This will break once the ES9 changes make their way to the mono-repo, I am assuming this entire client will need to be re-written.

Reply:
Yes, it would. It has already been rewritten by @KarpovIvan in the Optimize repo, we just need to integrate those changes here. I can assist @KarpovIvan when we get that far

Reply:
I don't think so, this is rebased against main. But because we reintroduce the ES RHLC again here, nothing breaks. Instead we need a follow up task to migrate this file to the new Java client

Reply:
Created https://github.com/camunda/camunda/issues/22518